Editor/Utility/CinemachineSceneTools.cs

```diff
@@ -0,0 +1,518 @@
+            fontStyle = FontStyle.Bold,
+            padding = new RectOffset(5, 0, 5, 0)
+        };
+        
+
+        public static float SliderHandleDelta(Vector3 newPos, Vector3 oldPos, Vector3 forward)
+        {
+            var delta = newPos - oldPos;
+            var sameDirection = Vector3.Dot(delta.normalized, forward) > 0;
+            return (sameDirection ? 1f : -1f) * delta.magnitude;
+        }
+
+        /// <summary>
+        /// Calculate delta and discard imprecision.
+        /// </summary>
+        public static Vector3 PositionHandleDelta(Quaternion rot, Vector3 newPos, Vector3 oldPos)
+        {
+            var delta =
+                Quaternion.Inverse(rot) * (newPos - oldPos);
+            delta = new Vector3(
+                Mathf.Abs(delta.x) < UnityVectorExtensions.Epsilon ? 0 : delta.x,
+                Mathf.Abs(delta.y) < UnityVectorExtensions.Epsilon ? 0 : delta.y,
+                Mathf.Abs(delta.z) < UnityVectorExtensions.Epsilon ? 0 : delta.z);
+            return delta;
+        }
+        
+        public static void DrawLabel(Vector3 position, string text)
+        {
+            var labelOffset = HandleUtility.GetHandleSize(position) / 5f;
+            Handles.Label(position + new Vector3(0, -labelOffset, 0), text, s_LabelStyle);
+        }
+
+        static int s_ScaleSliderHash = "ScaleSliderHash".GetHashCode();
+        static float s_FOVAfterLastToolModification;
+
+        public static void FovToolHandle(CinemachineVirtualCameraBase vcam, SerializedProperty lensProperty,
+            in LensSettings lens, bool isLensHorizontal)
+        {
+            var orthographic = lens.Orthographic;
+            if (GUIUtility.hotControl == 0)
+            {
+                s_FOVAfterLastToolModification = orthographic ? lens.OrthographicSize : lens.FieldOfView;
+            }
+            
+            var camPos = vcam.State.FinalPosition;
+            var camRot = vcam.State.FinalOrientation;
+            var camForward = camRot * Vector3.forward;
+                
+            EditorGUI.BeginChangeCheck();
+            var fovHandleId = GUIUtility.GetControlID(s_ScaleSliderHash, FocusType.Passive) + 1; // TODO: KGB workaround until id is exposed
+            var newFov = Handles.ScaleSlider(
+                s_FOVAfterLastToolModification, 
+                camPos, camForward, camRot, HandleUtility.GetHandleSize(camPos), 0.1f);
+            if (EditorGUI.EndChangeCheck())
+            {
+                if (orthographic)
+                {
+                    lensProperty.FindPropertyRelative("OrthographicSize").floatValue += 
+                        (s_FOVAfterLastToolModification - newFov);
+                }
+                else
+                {
+                    lensProperty.FindPropertyRelative("FieldOfView").floatValue += 
+                        (s_FOVAfterLastToolModification - newFov);
+                }
+                lensProperty.serializedObject.ApplyModifiedProperties();
```

Comment:
Undo isn't working for any of the lens tools.
Maybe because lensProperty.serializedObject gives you only a local copy of a lens, and not the owner object?

Reply:
Hmm... it works for me.